### PR TITLE
Function argument and plotting changes

### PR DESCRIPTION
**Pull request by jwbishop**
Updates the order of ltsva output. Updates plotting for optional station dictionary input (addresses #5 ). Adjusts example to recommended default alpha = 0.75.

### CODE REVIEW

**Comment by davidfee5** [comment]
lts
Essentially no elements are flagged for the example with alpha=0.75. Is this correct? I'm surprised.

File: lts_array/lts_array_plot.py [hunk 1]
```diff
@@ -29,86 +29,101 @@ def lts_array_plot(st, lts_vel, lts_baz, t, mdccm, stdict):
...
+    axs[0].text(0.15, 0.93, st[0].stats.station, horizontalalignment='center',
+                verticalalignment='center', transform=axs[0].transAxes)
+    cbaxes = fig.add_axes(
+        [0.95, axs[splot].get_position().y0, 0.02,
+         axs[vplot].get_position().y1 - axs[splot].get_position().y0])
 
     # Plot the trace velocity plot.
-    sc = axs1[1].scatter(t, lts_vel, c=mdccm,
-                         edgecolors='gray', lw=0.1, cmap=CM)
-    axs1[1].set_ylim(0.15, 0.60)
```
Comment (davidfee5) [inline lts_array/lts_array_plot.py:58]: Why 0.15-0.6? Anything other than ~0.25-0.45 is not realistic.

**Comment by davidfee5** [comment]
When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".

**Review by liamtoney — APPROVED** [review]

**Comment by jwbishop** [comment]
> When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".

This was actually done on purpose to preserve the output types, i.e. a dictionary is consistently returned. In the case of ordinary least squares, it's an empty dictionary. As in the array_processing plotting script, I expect the plotting default stdict=None behavior to be used when the station pair plot is undesired. For least squares, the output stdict variable can be ignored entirely by assigning it to `__` in the tuple unpacking.

Does that make sense?

**Comment by davidfee5** [comment]
> > When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".
> 
> This was actually done on purpose to preserve the output types, i.e. a dictionary is consistently returned. In the case of ordinary least squares, it's an empty dictionary. As in the array_processing plotting script, I expect the plotting default stdict=None behavior to be used when the station pair plot is undesired. For least squares, the output stdict variable can be ignored entirely by assigning it to `__` in the tuple unpacking.
> 
> Does that make sense?

OK, but then `    if stdict is not None:` on line 87 of lts_array_plot.py does not pass and it still tries to plot the element pairs subplot. Try the example with alpha=1.0 and you'll see what I mean.

File: lts_array/lts_array_plot.py (hunk 1 above)
Comment (jwbishop) [reply lts_array/lts_array_plot.py]: > > > When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".
> > 
> > 
> > This was actually done on purpose to preserve the output types, i.e. a dictionary is consistently returned. In the case of ordinary least squares, it's an empty dictionary. As in the array_processing plotting script, I expect the plotting default stdict=None behavior to be used when the station pair plot is undesired. For least squares, the output stdict variable can be ignored entirely by assigning it to `__` in the tuple unpacking.
> > Does that make sense?
> 
> OK, but then ` if stdict is not None:` on line 87 of lts_array_plot.py does not pass and it still tries to plot the element pairs subplot. Try the example with alpha=1.0 and you'll see what I mean.

Are you still passing a station dictionary as an argument to the plotting script? That will not evaluate as `None`, so the plotting script will attempt to plot the empty dictionary. I can fix this by adding an additional check for empty dictionaries. This should be done for the `array_processing` plotting script as well.

**Comment by jwbishop** [comment]
> > > When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".
> > 
> > 
> > This was actually done on purpose to preserve the output types, i.e. a dictionary is consistently returned. In the case of ordinary least squares, it's an empty dictionary. As in the array_processing plotting script, I expect the plotting default stdict=None behavior to be used when the station pair plot is undesired. For least squares, the output stdict variable can be ignored entirely by assigning it to `__` in the tuple unpacking.
> > Does that make sense?
> 
> OK, but then ` if stdict is not None:` on line 87 of lts_array_plot.py does not pass and it still tries to plot the element pairs subplot. Try the example with alpha=1.0 and you'll see what I mean.

Are you still passing a station dictionary as an argument to the plotting script? That will not evaluate as None, so the plotting script will attempt to plot the empty dictionary. I can fix this by adding an additional check for empty dictionaries. This should be done for the array_processing plotting script as well.

**Comment by jwbishop** [comment]
> > > > When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".
> > > 
> > > 
> > > This was actually done on purpose to preserve the output types, i.e. a dictionary is consistently returned. In the case of ordinary least squares, it's an empty dictionary. As in the array_processing plotting script, I expect the plotting default stdict=None behavior to be used when the station pair plot is undesired. For least squares, the output stdict variable can be ignored entirely by assigning it to `__` in the tuple unpacking.
> > > Does that make sense?
> > 
> > 
> > OK, but then ` if stdict is not None:` on line 87 of lts_array_plot.py does not pass and it still tries to plot the element pairs subplot. Try the example with alpha=1.0 and you'll see what I mean.
> 
> Are you still passing a station dictionary as an argument to the plotting script? That will not evaluate as None, so the plotting script will attempt to plot the empty dictionary. I can fix this by adding an additional check for empty dictionaries. This should be done for the array_processing plotting script as well.

Fixed. This was actually an issue where `ltsva` was not returning a truly empty dictionary for least squares processing.

File: lts_array/lts_array_plot.py (hunk 1 above)
Comment (jwbishop) [reply lts_array/lts_array_plot.py:58]: Fixed!

File: lts_array/lts_array_plot.py [hunk 2]
```diff
@@ -29,86 +29,105 @@ def lts_array_plot(st, lts_vel, lts_baz, t, mdccm, stdict):
...
-    axs1[1].grid(b=1, which='major', color='gray', linestyle=':', alpha=0.5)
+    sc = axs[vplot].scatter(t, lts_vel, c=mdccm,
+                            edgecolors='gray', lw=0.1, cmap=cm)
+    axs[vplot].set_ylim(0.25, 0.45)
+    axs[vplot].set_xlim(t[0], t[-1])
+    axs[vplot].plot([t[0], t[-1]], [0.25, 0.25], '-', color='grey')
+    axs[vplot].plot([t[0], t[-1]], [0.45, 0.45], '-', color='grey')
+    sc.set_clim(cax)
+    axs[vplot].set_ylabel('Trace Velocity\n [km/s]')
+    axs[vplot].grid(b=1, which='major', color='gray', linestyle=':', alpha=0.5)
```
Comment (davidfee5) [inline lts_array/lts_array_plot.py]: One more: why do you have the gridlines gray? It's different than the top subplot and how we usually plot

**Review by davidfee5 — APPROVED** [review]

**Review by liamtoney — APPROVED** [review]

**Pull request merged**